src/plugins/analysis/ipc/internal/ipcAnalyzer/ipcAnalyzer.py

```diff
@@ -0,0 +1,380 @@
+        else:
+            # Case if the string is less than 5 charactes long
+            try:
+                byte = getByte(addr)
+                cString = ""
+                for i in range(6):
+                    if byte == 0:
+                        break
+                    cString += chr(byte)
+                    i += 1
+                    byte = getByte(addr.add(i))
+                return cString.encode('utf-8').strip()
+            except:
+                return data
+    return data
+
+def parseOpenFlags(data):
+    """
+    Tries to parse open flags
+
+    :param data: long/list
+    :return: str
+    """
+    if isinstance(data, list):
+        return forEachData(data, parseOpenFlags)
+    symbols = openflags2symbols(data)
+    return symbols
+
+def parseOpenMode(data):
+    """
+    Tries to parse open mode
+
+    :param data: long/list
+    :return: str
+    """
+    if isinstance(data, list):
+        return forEachData(data, parseOpenMode)
+    if isinstance(data, (int, long)):
+        return oct(data)
+    return data
+
+def addJsonCall(ipc, functionName, argValues):
+    """
+    Adds an ipc call to the JSON file
+
+    :param ipc: dict
+    :param functionName: unicode
+    :param argValues: list
+    :return: None
+    """
+    firstArg = argValues[0]
+    if len(argValues) > 1:
+        restArgs = argValues[1:]
+    else:
+        restArgs = []
+    if isinstance(firstArg, list):
+        for arg in firstArg:
+            addJsonCall(ipc, functionName, [arg]+restArgs)
+    elif firstArg is not None and isinstance(firstArg, str) and len(firstArg) >= 1 and firstArg[0] not in ['%', '?', '0']:
+        target = firstArg.split()[0]
+        if target not in ipc["ipcCalls"]:
+            ipc["ipcCalls"][target] = []
+        ipcCall = {"type": functionName}
+        ipcCall["arguments"] = [" ".join(firstArg.split()[1:])] + restArgs
+        ipc["ipcCalls"][target].append(ipcCall)
+
+def writeToFile(outputFile, result, resultPath='.'):
+    """
+    Writes the json to file
+
+    :param outputFile: dict
+    :param result: str
+    :param resultPath: unicode
+    :return: None
+    """
+    print("\nWriting {}".format(resultPath + '/' + outputFile))
+    with open(resultPath + '/' + outputFile, 'wb') as f:
+        json.dump(result, f, indent=2)
+    os.chmod(resultPath + '/' + outputFile, 0o666)  # assure access rights to file created in docker container
+
+def isCorrect(argValue):
+    """
+    Checks if the argument only contains strings
+
+    :param argValue: list, str, int, long
+    :return: bool
+    """
+    result = True
+    if isinstance(argValue, list):
+        for arg in argValue:
+            result = result and isCorrect(arg)
+        return result
+    else:
+        if argValue is None or isinstance(argValue, (int, long)):
+            return False
+        else:
+            return True
+
+def resolveVersionFormatString(ghidraAnalysis, keyStringList):
```

Comment:
This function is really long and has a high nesting depth. It would probably be good idea to move some parts into functions to improve clarity.

Reply:
I have divided the function into several parts.